ludwig/utils/config_utils.py

```diff
@@ -68,37 +66,6 @@ def merge_config_preprocessing_with_feature_specific_defaults(
     return preprocessing_params
 
 
-@DeveloperAPI
-def merge_fixed_preprocessing_params(
-    feature_type: str, preprocessing_params: Dict[str, Any], encoder_params: Dict[str, Any]
-) -> Dict[str, Any]:
-    """Update preprocessing parameters if encoders require fixed preprocessing parameters."""
-    encoder_type = encoder_params.get(TYPE, get_default_encoder_type(feature_type))
-    encoder_class = get_encoder_cls(feature_type, encoder_type)
-    return merge_dict(preprocessing_params, encoder_class.get_fixed_preprocessing_params(encoder_params))
-
-
-@DeveloperAPI
-def get_default_encoder_type(feature_type: str) -> str:
-    feature_schema = get_from_registry(feature_type, get_input_type_registry()).get_schema_cls()
-    return feature_schema().encoder.type
-
-
-@DeveloperAPI
-def get_default_decoder_type(feature_type: str) -> str:
-    feature_schema = get_from_registry(feature_type, get_output_type_registry()).get_schema_cls()
-    return feature_schema().decoder.type
-
-
-@DeveloperAPI
-def get_default_encoder_or_decoder(feature: FeatureConfigDict, config_feature_group: str) -> str:
-    """Returns the default encoder or decoder for a feature."""
-    if config_feature_group == INPUT_FEATURES:
-        return get_default_encoder_type(feature[TYPE])
-    else:
-        return get_default_decoder_type(feature[TYPE])
-
-
 def has_trainable_encoder(config: ModelConfig) -> bool:
```

Comment:
nit: would it make sense to also move `has_trainable_encoder` and `has_pretrained_encoder` into `ludwig/schema/model_types/utils.py`?

Reply:
Yes, we can do it in a follow-up.